CARDS-2000 - Patient portal: "Sign out" in the header looks confusing to patients when only token auth is enabled and the patient has not actively signed in

marta- opened the pull request:
- Removed the "Sign out" link when the patient wasn't required to enter their mrn&dob in the identification form before proceeding
- ~When the patient closes the page, automatically send a call to logout~

**Testing:**
* check that Sign out is no longer displayed for PREMs
* check that the header layout looks ok without that link

**Regression testing:**
* ensure the PROMs workflow (where the patient must sign in with mrn&dob) is not affected
* ensure the Sign Out link is still displayed for PROMs

marta- committed:
CARDS-2000 - Patient portal: "Sign out" in the header looks confusing to patients when only token auth is enabled and the patient has not actively signed in
- Removed the "Sign out" link when the patient wasn't required to enter their mrn&dob in the identification form before proceeding
- When the patient closes the page, automatically send a call to logout

sashaandjic commented:
In `PREMS` mode the only way to logout is after completing the survey, which is OK.
In `PROMS` mode after completing the last survey instead of getting the `REVIEW YOUR RESPONSES` I get this:
![image](https://user-images.githubusercontent.com/23243244/212412462-518dec5f-7174-47ec-98b8-fbab249e6875.png)

sashaandjic reviewed (approved):
tested - the problem previously reported is not related to this PR and it is logged [here](https://phenotips.atlassian.net/browse/CARDS-2005).

sdumitriu reviewed (commented):
The new logout request makes this a bit ugly.

- in prems, set up a new CPES visit, connect as a patient with the token, start filling in the form but don't get to the end
- try to close the page
- due to the multiple requests being sent, sometimes the browser's native login dialog shows up on top of the browser's "are you sure you want to leave" dialog
- it's also possible that if the user says no to the "are you sure you want to leave" question, continuing to fill in the form will bring up the login dialog since the user is logged out
- it's also possible, due to the multiple requests, that the logout failed and the patient will still be logged in

How do you want to handle this? Since the logout is kind of broken anyway, I would remove the new logout code from this PR, and try to handle it properly when we fix the multiple requests issue.

marta- commented:
> The new logout request makes this a bit ugly.
> 
> * in prems, set up a new CPES visit, connect as a patient with the token, start filling in the form but don't get to the end
> * try to close the page
> * due to the multiple requests being sent, sometimes the browser's native login dialog shows up on top of the browser's "are you sure you want to leave" dialog

Only in Firefox.

> * it's also possible that if the user says no to the "are you sure you want to leave" question, continuing to fill in the form will bring up the login dialog since the user is logged out
> * it's also possible, due to the multiple requests, that the logout failed and the patient will still be logged in
> 
> How do you want to handle this? Since the logout is kind of broken anyway, I would remove the new logout code from this PR, and try to handle it properly when we fix the multiple requests issue.

Ok.

marta- committed:
CARDS-2000 - Patient portal: "Sign out" in the header looks confusing to patients when only token auth is enabled and the patient has not actively signed in

Addressing review comments: removed the automated sign-out when the patient leaves the page
